File: Find second largest element from an array.cpp
int getSecondLargest(vector<int> &arr) {
    int n = arr.size();

    int largest = -1, secondLargest = -1;

    for (int i = 0; i < n; i++) {
        if(arr[i] > largest) {
            secondLargest = largest;
            largest = arr[i];
        }
        else if(arr[i] < largest && arr[i] > secondLargest) {
            secondLargest = arr[i];
        }
    }
    return secondLargest;
}

File: Find second largest element from an array.java
import java.util.List;

public class SecondLargest {

    public static int getSecondLargest(List<Integer> arr) {
        int n = arr.size();

      
        int largest = -1;
        int secondLargest = -1;

        
        for (int i = 0; i < n; i++) {
            if (arr.get(i) > largest) {
                secondLargest = largest;
                largest = arr.get(i);
            } else if (arr.get(i) < largest && arr.get(i) > secondLargest) {
                secondLargest = arr.get(i);
            }
        }
        return secondLargest;
    }

    public static void main(String[] args) {
        List<Integer> arr = List.of(12, 35, 1, 10, 34, 1);
        System.out.println("Second Largest: " + getSecondLargest(arr));
    }
}
